geomstats/special_euclidean_group.py

```diff
@@ -526,43 +574,49 @@ def group_exponential_barycenter(self, points, weights=None):
-
-        vec_aux = rotations.rotation_vector_from_matrix(matrix_aux)
-        matrix_aux = self.exponential_matrix(vec_aux)
-        matrix_aux = gs.linalg.inv(matrix_aux)
-
-        for i in range(n_points):
-            matrix += weights[i] * matrix_aux[i]
-            translation_aux += weights[i] * gs.dot(gs.matmul(
-                                                        matrix_aux[i],
-                                                        inv_rot_mats[i]),
-                                                   translations[i])
-
-        mean_translation = gs.dot(translation_aux,
-                                  gs.transpose(gs.linalg.inv(matrix),
-                                               axes=(0, 2, 1)))
-
-        exp_bar = gs.zeros((1, dim))
-        exp_bar[0, :dim_rotations] = mean_rotation
-        exp_bar[0, dim_rotations:dim] = mean_translation
+        if point_type == 'vector':
+            rotation_vectors = points[:, :dim_rotations]
+            translations = points[:, dim_rotations:dim]
+            assert rotation_vectors.shape == (n_points, dim_rotations)
+            assert translations.shape == (n_points, self.n)
+
+            mean_rotation = rotations.group_exponential_barycenter(
+                                                    points=rotation_vectors,
+                                                    weights=weights)
+            mean_rotation_mat = rotations.matrix_from_rotation_vector(
+                        mean_rotation)
+
+            matrix = gs.zeros((1,) + (self.n,) * 2)
+            translation_aux = gs.zeros((1, self.n))
+
+            inv_rot_mats = rotations.matrix_from_rotation_vector(
+                    -rotation_vectors)
+            # TODO(nina): this is the same mat multiplied several times
+            matrix_aux = gs.matmul(mean_rotation_mat, inv_rot_mats)
+            assert matrix_aux.shape == (n_points,) + (dim_rotations,) * 2
+
+            vec_aux = rotations.rotation_vector_from_matrix(matrix_aux)
+            matrix_aux = self.exponential_matrix(vec_aux)
+            matrix_aux = gs.linalg.inv(matrix_aux)
+
+            for i in range(n_points):
+                matrix += weights[i] * matrix_aux[i]
+                translation_aux += weights[i] * gs.dot(gs.matmul(
+                                                            matrix_aux[i],
+                                                            inv_rot_mats[i]),
+                                                       translations[i])
+
+            mean_translation = gs.dot(translation_aux,
+                                      gs.transpose(gs.linalg.inv(matrix),
+                                                   axes=(0, 2, 1)))
+
+            exp_bar = gs.zeros((1, dim))
+            exp_bar[0, :dim_rotations] = mean_rotation
+            exp_bar[0, dim_rotations:dim] = mean_translation
 
+        elif point_type == 'matrix':
+            points = self.rotation_vector_from_matrix(points)
```

Comment:
for more clarity - maybe call this one vector_points or vectors? That way the next method is more readable.